### PR TITLE
Include permissions & view permissions in table perspective #LMR-1197

### PR DESCRIPTION
**Pull request by kubedo8**

### CODE REVIEW

File: src/app/core/model/allowed-permissions.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+export interface AllowedPermissions {
+  readWithView?: boolean;
+  writeWithView?: boolean;
+  manageWithView?: boolean;
+  read?: boolean;
+  write?: boolean;
+  manage?: boolean;
```
[the remark this answers is not in the document]
Comment (kubedo8) [reply src/app/core/model/allowed-permissions.ts]: Sometimes you need to know if you have direct access to data or through the view.

File: src/app/core/store/tables/table.utils.ts [hunk 2]
```diff
@@ -294,23 +294,23 @@ export function resizeLastColumnChild(column: TableCompoundColumn, delta: number
 
 export const HIDDEN_COLUMN_WIDTH = 10;
 
-export function getTableColumnWidth(column: TableColumn): number {
+export function getTableColumnWidth(column: TableColumn, showHiddenColumns: boolean): number {
```
[the remark this answers is not in the document]
Comment (kubedo8) [reply src/app/core/store/tables/table.utils.ts]: I think that yes. What do you think about it @marvec ?

File: src/app/shared/pipes/pipes.module.ts [hunk 3]
```diff
@@ -43,9 +43,13 @@ import {UserRolesInResource} from './user-roles-in-resource.pipe';
...
 import {CanActivatePagePipe} from './can-activate-page.pipe';
 import {PageEndIndexPipe} from './page-end-index.pipe';
-import { LogPipe } from './log.pipe';
-import { RemoveHtmlCommentsPipe } from './remove-html-comments.pipe';
+import {LogPipe} from './log.pipe';
+import {RemoveHtmlCommentsPipe} from './remove-html-comments.pipe';
 import {CollectionPermissionsPipe} from './collection-permissions.pipe';
+import {LinkTypePermissionsPipe} from './link-type-permissions.pipe';
+import {DocumentPermissionsPipe} from './document-permissions.pipe';
+import {LinkInstancePermissionsPipe} from './link-instance-permissions.pipe';
```
[the remark this answers is not in the document]
Comment (kubedo8) [reply src/app/shared/pipes/pipes.module.ts]: Okay

**Comment by kubedo8** [comment]
@livthomas Now it should be okay :)

File: src/app/core/store/tables/table.utils.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (marvec) [reply src/app/core/store/tables/table.utils.ts]: Yes!

**Pull request merged**